client/src/main/java/org/apache/cassandra/sidecar/client/SidecarClient.java

```diff
@@ -489,6 +492,45 @@ public CompletableFuture<Void> cleanUploadSession(SidecarInstance instance, Stri
                                                             .build());
     }
 
+    /**
+     * Lists CDC commit logs in CDC directory for an instance
+     * @param sidecarInstance instance on which the CDC commit logs are to be listed
+     * @return a completable future with List of cdc commitLogs on the requested instance
+     */
+    public CompletableFuture<ListCdcSegmentsResponse> listCdcSegments(SidecarInstance sidecarInstance)
+    {
+        return executor.executeRequestAsync(requestBuilder()
+                .singleInstanceSelectionPolicy(sidecarInstance)
+                .request(new ListCommitLogsRequest())
+                .build());
+    }
+
+    /**
+     *  Streams CDC commit log segments from the requested instance.
+     *
+     * Streams the specified {@code range} of a CDC CommitLog from the given instance and the
+     * stream is consumed by the {@link StreamConsumer consumer}.
+     *
+     * @param sidecarInstance instance on which the CDC commit logs are to be streamed
+     * @param segment segment file name
+     * @param range range of the file to be streamed
+     * @param partitionId partition ID
+     * @param batchId batchId
+     * @param streamConsumer object that consumes the stream
+     */
+    public void streamCommitLogs(SidecarInstance sidecarInstance,
+                                 String segment,
+                                 HttpRange range,
+                                 String partitionId,
+                                 String batchId,
```

Comment:
Not sure if we require PartitionId and BatchId, they are not used by the API but I put them here as we might need them for metrics or logging purpose.

Reply:
Maybe having them as optionals? (that would require checking them not to be null before setting headers)

Reply:
I would avoid using Optionals here. If we don't need it for this PR, maybe avoid adding it?